### ISSUE TITLE
Passenger standalone built-in engine broken in 5.1.5?

### ISSUE THREAD

**Issue by karlsvec** [phusion/passenger#1963]
My system:

* Ubuntu Linux 14.04.5 LTS
* Passenger standalone 5.1.5, using the built-in engine
* Passenger installed via the Phusion APT repository
* Ruby 2.3.4p301 from the [Brightbox PPA](https://launchpad.net/~brightbox/+archive/ubuntu/ruby-ng)

Since upgrading to Passenger 5.1.5, my Rails application no longer starts.

My app and system configuration works on 5.1.2 and 5.1.3, so I suspect that this is a regression in Passenger.

I'm using Upstart to manage Passenger. Here's my configuration:

```
description "Passenger web application server"

author "Karl Svec"

start on (local-filesystems and net-device-up IFACE!=lo)
stop on stopping network-services

respawn

console log

env PASSENGER_COMPILE_NATIVE_SUPPORT_BINARY=0
env PASSENGER_DOWNLOAD_NATIVE_SUPPORT_BINARY=0

pre-start script
    if [ ! -d "/var/run/passenger" ]
    then
        install -o nobody -g www-data -m 750 -d /var/run/passenger
    fi
end script

exec start-stop-daemon --start \
                       --chuid nobody \
                       --chdir /opt/agsolver/pzm/layer/webapp/deploy/current \
                       --pidfile /var/run/passenger/passenger.pid \
                       --exec /usr/bin/ruby2.3 \
                       -- /usr/bin/passenger start \
                       --engine builtin \
                       --socket /var/run/passenger/passenger.sock \
                       --min-instances 0 \
                       --max-pool-size 5 \
                       --max-preloader-idle-time 0 \
                       --max-requests 500 \
                       --pool-idle-time 60 \
                       --log-file /tmp/passenger.log \
                       --pid-file /var/run/passenger/passenger.pid \
                       --environment web_development \
                       --no-friendly-error-pages \
                       --ruby /opt/agsolver/bin/passenger_ruby.sh
```

**NOTE**: `/opt/agsolver/bin/passenger_ruby.sh` is a "pass-through" Bash script that just sets a bunch of environment variables to configure the app and finally executes:

```sh
exec /usr/bin/ruby2.3 "$@"
```

The log file at `/tmp/passenger.log` contains the following message indicating that the preloader process died:

```
[ 2017-07-19 19:53:27.7952 8874/7f3ae4ebe700 age/Cor/App/Implementation.cpp:304 ]: Could not spawn process for application /opt/agsolver/pzm/layer/webapp/deploy/releases/190470277cd59e6b988f8c16fe72033ebc0b5f45: An error occurred while starting up the preloader. It exited before signalling successful startup back to Phusion Passenger.
```

Turning up the Passenger log level does not yield any additional information.

I can't even get a dummy Rack app to work on 5.1.5 with the built-in engine on a fresh Ubuntu 14.04 VM:

**`config.ru`**:
```ruby
run Proc.new { |env| ['200', {'Content-Type' => 'text/html'}, ['get rack\'d']] }
```

Starting the dummy app using the built-in engine does not work:

```
$ /usr/bin/ruby2.3 /usr/bin/passenger start --app-type rack --startup-file config.ru --engine builtin
 [passenger_native_support.so] trying to compile for the current user (vagrant) and Ruby interpreter...
     (set PASSENGER_COMPILE_NATIVE_SUPPORT_BINARY=0 to disable)
     Warning: compilation didn't succeed. To learn why, read this file:
     /tmp/passenger_native_support-2urns0.log
 [passenger_native_support.so] finding downloads for the current Ruby interpreter...
     (set PASSENGER_DOWNLOAD_NATIVE_SUPPORT_BINARY=0 to disable)
     Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 404 Not Found
     Trying next mirror...
     Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 403 Forbidden
 [passenger_native_support.so] will not be used (can't compile or download)
  --> Passenger will still operate normally.
=============== Phusion Passenger Standalone web server started ===============
PID file: /home/vagrant/passenger.3000.pid
Log file: /home/vagrant/passenger.3000.log
Environment: development
Accessible via: http://0.0.0.0:3000/

You can stop Phusion Passenger Standalone by pressing Ctrl-C.
Problems? Check https://www.phusionpassenger.com/library/admin/standalone/troubleshooting/
===============================================================================
[ 2017-07-19 20:29:57.3669 2464/7f87f817e700 age/Cor/Con/CheckoutSession.cpp:315 ]: [Client 1-1] Cannot checkout session due to Passenger::RuntimeException: No startCommand given
     in 'virtual Passenger::SpawningKit::Result Passenger::SpawningKit::DirectSpawner::spawn(const Options&)' (DirectSpawner.h:162)
     in 'void Passenger::ApplicationPool2::Group::spawnThreadRealMain(const SpawnerPtr&, const Passenger::ApplicationPool2::Options&, unsigned int)' (SpawningAndRestarting.cpp:95)

[ 2017-07-19 20:29:58.0587 2464/7f87ff0c5700 age/Cor/SecurityUpdateChecker.h:356 ]: Security update check: no update found (next check in 24 hours)
```

Passenger complains with "`no startCommand given`".

In contrast, the Nginx engine _does_ work:

```
$ /usr/bin/ruby2.3 /usr/bin/passenger start --app-type rack --startup-file config.ru --engine nginx
 [passenger_native_support.so] trying to compile for the current user (vagrant) and Ruby interpreter...
     (set PASSENGER_COMPILE_NATIVE_SUPPORT_BINARY=0 to disable)
     Warning: compilation didn't succeed. To learn why, read this file:
     /tmp/passenger_native_support-1x2t8ac.log
 [passenger_native_support.so] finding downloads for the current Ruby interpreter...
     (set PASSENGER_DOWNLOAD_NATIVE_SUPPORT_BINARY=0 to disable)
     Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 404 Not Found
     Trying next mirror...
     Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 403 Forbidden
 [passenger_native_support.so] will not be used (can't compile or download)
  --> Passenger will still operate normally.
=============== Phusion Passenger Standalone web server started ===============
PID file: /home/vagrant/passenger.3000.pid
Log file: /home/vagrant/passenger.3000.log
Environment: development
Accessible via: http://0.0.0.0:3000/

You can stop Phusion Passenger Standalone by pressing Ctrl-C.
Problems? Check https://www.phusionpassenger.com/library/admin/standalone/troubleshooting/
===============================================================================
App 2565 stdout:
App 2565 stderr:  [passenger_native_support.so] trying to compile for the current user (vagrant) and Ruby interpreter...
App 2565 stderr:      (set PASSENGER_COMPILE_NATIVE_SUPPORT_BINARY=0 to disable)
App 2565 stderr:      Warning: compilation didn't succeed. To learn why, read this file:
App 2565 stderr:      /tmp/passenger_native_support-smfl18.log
App 2565 stderr:  [passenger_native_support.so] finding downloads for the current Ruby interpreter...
App 2565 stderr:      (set PASSENGER_DOWNLOAD_NATIVE_SUPPORT_BINARY=0 to disable)
[ 2017-07-19 20:34:39.1628 2527/7f882b5ec700 age/Cor/SecurityUpdateChecker.h:356 ]: Security update check: no update found (next check in 24 hours)
App 2565 stderr:      Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 404 Not Found
App 2565 stderr:      Trying next mirror...
App 2565 stderr:      Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/5.1.5/rubyext-ruby-2.3.4-x86_64-linux.tar.gz: The requested URL returned error: 403 Forbidden
App 2565 stderr:  [passenger_native_support.so] will not be used (can't compile or download)
App 2565 stderr:   --> Passenger will still operate normally.
App 2590 stdout:
```

```
$ curl http://localhost:3000
get rack'd
```

**Comment by CamJN** [comment]
Yeah we are aware of this and it is fixed in 5.1.6 which is coming soon. Sorry about that.

**Comment by OnixGH** [comment]
Relevant commit it was fixed in: https://github.com/phusion/passenger/commit/4f09fb9d411256433907fa8a96b9827bcc519488
